drivers: bluetooth: hci: Simplify SPI driver

HoZHel opened the pull request:
Used `LOG_HEXDUMP_DBG` for SPI message logging instead of `printk`.

Removed redundant `configure_cs` as it is performed by SPI initialization.

Modified SPI configuration to match the features introduced in PR #63437.

erwango reviewed (changes requested):
Please split changes into logical commits (boards config update, `LOG_HEX_DUMP` introduction, `configure_cs` removal, ....)

HoZHel committed:
drivers: bluetooth: hci: Replace printk with LOG_HEXDUMP_DBG

Use LOG_HEXDUMP_DBG for SPI message logging instead of printk.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

JordanYates commented on drivers/bluetooth/hci/spi.c:
```diff
@@ -304,8 +264,8 @@ static struct net_buf *bt_spi_rx_buf_construct(uint8_t *msg)
-			if (msg[1] == BT_HCI_EVT_LE_META_EVENT &&
-			    (msg[3] == BT_HCI_EVT_LE_ADVERTISING_REPORT)) {
+			if (msg[EVT_HEADER_EVENT] == BT_HCI_EVT_LE_META_EVENT &&
+			    (msg[EVT_VENDOR_CODE_LSB] == BT_HCI_EVT_LE_ADVERTISING_REPORT)) {
```
This is not a vendor event code, so should not be using vendor event code defines

HoZHel replied:
Done.

HoZHel committed:
drivers: bluetooth: hci: Omit configure_cs

Remove redundant configure_cs as it is performed by SPI initialization.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

HoZHel committed:
drivers: bluetooth: hci: SPI configuration

Modify SPI configuration to match the features introduced in PR #63437.

Set the property "controller-data-delay-us" to zero for boards
which are using BlueNRG-MS.

Fix Chip Select configuration for stm32l562e_dk board.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

HoZHel committed:
samples: bluetooth: hci_usb: Test exclusion for stm32l562e_dk

STM32L562 Discovery kit is not compatible with Bluetooth over USB
application.

Add missing information related to the BLE capability of this board.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

HoZHel committed:
drivers: bluetooth: hci: Use macros instead of hard-coding

Replace hard-coded values with suitable macros.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

erwango commented on boards/arm/stm32l562e_dk/stm32l562e_dk_common.dtsi:
```diff
@@ -111,7 +111,7 @@
 	pinctrl-0 = <&spi1_sck_pg2 &spi1_miso_pg3 &spi1_mosi_pg4>;
 	pinctrl-names = "default";
-	cs-gpios = <&gpiog 5 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+	cs-gpios = <&gpiog 5 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
```
Non blocking: A word on this change in commit message wouldn't harm.

HoZHel replied:
Done.

erwango commented on boards/arm/stm32l562e_dk/stm32l562e_dk_common.dtsi:
```diff
@@ -120,6 +120,8 @@
 		irq-gpios = <&gpiog 6 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>;
 		reset-gpios = <&gpiog 8 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
 		spi-max-frequency = <2000000>;
+		controller-data-delay-us = <0>;  /* No need for extra delay for BlueNRG-MS */
```
Ditto

HoZHel replied:
Done.

erwango reviewed (commented):
2 minor comments, otherwise LGTM.
BLE maintainer approval required anyway

alwa-nordic commented on samples/bluetooth/hci_usb/sample.yaml:
```diff
@@ -10,7 +10,7 @@ tests:
       - bluetooth
     # FIXME: exclude due to build error
-    platform_exclude: 96b_carbon
+    platform_exclude: 96b_carbon stm32l562e_dk
```
Is this because of USB?

HoZHel replied:
Yes. Since the platform is running a 2-chip configuration where the controller part is running on BlueNRG-MS using an SPI interface, and the STM32L4xxxx is running the host part connected via SPI to the controller, therefore, the hci_usb protocol does not apply to this platform.

avisconti reviewed (approved):
LGTM :)
